torch/nn/utils/spectral_norm.py

```diff
@@ -51,6 +51,8 @@ def __call__(self, module, inputs):
             weight, u = self.compute_weight(module)
             setattr(module, self.name, weight)
             setattr(module, self.name + '_u', u)
+        else:
+            getattr(module, self.name).detach_()
```

Comment:
This should have `.requires_grad_()`. Arguably the weight should still be backprop-able in eval mode.

Reply:
But is it OK to reset this for each iteration or should I set to the previous grad (or that of weight_orig)?

Reply:
Set previous `requires_grad` is definitely better! Go for it.

Reply:
`weight`'s `requires_grad` should be the same as that of `weight_orig`, right?

Reply:
Well, only `weight_orig` shows up in the parameters, so this situation might not be too uncommon:
I'm looking at this from the perspective of my (possibly not terribly good) [SN-GAN implementation](https://nbviewer.jupyter.org/github/t-vi/pytorch-tvmisc/blob/master/wasserstein-distance/sn_projection_cgan_64x64_1000c.ipynb)
For the generator update I do
```
discriminator.eval()
for p in discriminator.parameters():
   p.requires_grad = False
```
this sees weight_orig, but not weight. So weight_orig will be `requires_grad = False` and weight `requires_grad = True`. One might argue that setting weight's requires_grad to weight_orig's is less surprising here.

Reply:
So I changed it to weight_orig's requires grad, I hope that's OK.

Reply:
Yeah it should be okay. My argument was that since `weight` is computed from `weight_orig` (in most cases), it should have the same `requires_grad` flag.